keps/sig-architecture/3138-increase-reliability-bar/README.md

```diff
@@ -0,0 +1,415 @@
+reliability and our velocity medium and longer term.
+
+To mitigate the risk, we will provide the necessary tooling to ensure that any
+single SIG will not get surprised by the consequences of conscious
+deprioritization of reliability-related work.**
+
+
+Another risk is that due to pulling engineers away from the most interesting
+feature work towards non-fancy work, we may see attritions and decrease in
+contributor base. This is why it is critical to actually measure the benefits
+from this effort. Most of the metrics will turn out to be lagging indicators,
+so they will rather be used to measure the impact and reflect on the process
+periodically, not necessarily guide immediate future actions.
+
+The exact metrics are to-be-decided, but the current candidates include:
+- number of bugs (ideally weighed by severity) reported per SIG for already
+  released versions:
+  - for cross-cutting issues, they will be counted towards “owning SIG”
+- number of cherrypicks per SIG for a given minor release (targeting low
+  numbers meaning that SIG doesn’t have any issues that need to be addressed).
+  - for cross-cutting fixes, they will be counted towards “owning SIG”
+- test flakiness (as observed in https://sippy.k8s.io/?release=kube-master)
+- test coverage (the details of how to measure it are to-be-decided)
+- [ideal but unlikely due to inability to get the data] number of outages
+  reported by Kubernetes providers
+
+
+## Design Details
+
+As mentioned above, we propose to divide the effort into 4 milestones.
+Below we describe these milestones, focusing on the first milestone that
+will be executed first as a pilot.
+
+### [Milestone 1] Existing tests failures/flakiness
+
+When a test fails (especially the end-to-end test), the reason for the failure
+is often unclear. All of the following reasons are theoretically possible:
+- test-related issue (e.g. test written in non-deterministic way, incorrect
+  CI job configuration, etc.)
+- a bug in the exercised component
+- a bug in another component (potentially owned by a different SIG)
+- some underlying infrastructure-related issues (e.g. cloud-provider outage)
+
+For sustainability, we should figure out a path towards understanding them.
+Obviously we’re extremely far from being able to do that now, but without any
+work the amount of necessary work to get there can only increase - it will
+not decrease on its own.
+
+To make the effort trackable and sustainable we propose the following:
+1. We will limit the effort only to the test jobs that are part of
+   “master release blocking” and “master release informing” tabs.
+   - If really needed we may consider extending the set of jobs in scope,
+     but this is not a plan of record as of now.
+
+1. We will use [Sippy][] as our source of truth for determining the health
+   of the individual jobs and tests.
+   We’ve already set up [Kubernetes Sippy][] that is monitoring the jobs
```

Comment:
we have to make some commitment the tool will be maintained and what sig is going to own it

Reply:
Agree.

Reply:
It sounds like a prereq for this work or step 0 is to solidify ownership of the tools and the jobs (which was also [noted below](https://github.com/kubernetes/enhancements/pull/3139/files#r807860207)).